### PR TITLE
Move vs15.9 branch to AzDO

### PR DESCRIPTION
**Pull request by mmitche**

### CODE REVIEW

**Comment by mmitche** [comment]
@rainersigwald What do you thiink we should do with the RHEL leg? Is it needed for 15.9 validation? If it is, should we try to build it in the same docker container that is used to build source build. The hosted pool doesn't have RHEL.

**Comment by rainersigwald** [comment]
I'm comfortable dropping the RHEL leg. It was a "source-build-like" thing, but I don't th ink it was ever all that useful.

File: .vsts-dotnet-ci.yml [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+- vs15.9
+
+pr:
+- vs15.9
+
+jobs:
+- job: innerloop_OSX1013_CoreCLR
+  displayName: OSX10.13 Build for CoreCLR
+  pool:
+    vmImage: macOS-10.13
```
Comment (chcosta) [inline .vsts-dotnet-ci.yml]: nit: would the "Hosted macOS High Sierra" pool work for this?  It would be nice to move to hosted pools where possible

Comment (mmitche) [reply .vsts-dotnet-ci.yml]: This is the hosted pool.  They wanted us to use this format (the other format is deprecated)

Comment (mmitche) [reply .vsts-dotnet-ci.yml]: https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops#use-a-microsoft-hosted-agent

Comment (chcosta) [reply .vsts-dotnet-ci.yml]: boom

**Comment by rainersigwald** [comment]
To get tests green on Pipelines I think you'll need to cherry-pick some subset of https://github.com/microsoft/msbuild/pull/4248. Probably 917b7a5, 
a9bd6b2, and 1b9c27d.

**Comment by mmitche** [comment]
@rainersigwald Can you take a look at those test failures when you get a chance? Given that these are stock hosted pool machines, I would not have expected that.

**Comment by rainersigwald** [comment]
Yeah, that's what I think needs the cherry-picks I mentioned above. Pipelines flipped the long-path support bit on their hosted machines at some point.

**Comment by mmitche** [comment]
> Yeah, that's what I think needs the cherry-picks I mentioned above. Pipelines flipped the long-path support bit on their hosted machines at some point.

Ahh sorry missed that comment.

**Comment by rainersigwald** [comment]
Ok, this failure is new:

```
Assert.True() Failure
Expected: True
Actual: False

   at Microsoft.Build.UnitTests.GenerateResource_Tests.Utilities.ExecuteTask(GenerateResource t) in /_/src/Tasks.UnitTests/GenerateResource_Tests.cs:line 3306
   at Microsoft.Build.UnitTests.GenerateResource_Tests.InProc.RequiredTransformations.ForceSomeOutOfDate() in /_/src/Tasks.UnitTests/GenerateResource_Tests.cs:line 435
```

**Pull request merged**